### ISSUE TITLE
R4i Gold 3DS Rev 7 (www.r4ids.cn)

### ISSUE THREAD

**Issue by Leadger69** [kitling/flashcart_core#75]
I've dumped and flashed ntrboot but when I try to run it on my other 3ds it won't boot into safeboot.
But when I try the ntrboot flasher 3.0 it says that flashcart is not supported 

Newest wood kernal: https://filetrip.net/dl?FhXOBrY7vI
Newest upgrade patch: https://filetrip.net/dl?3mbPXwTcpP

ChipID: 00000FC2
HW Revision: 00000000

<details><summary>Images <em>(Click to Expand)</em></summary><p> 

https://imgur.com/gnhpU1Q
https://imgur.com/nMUax44
https://imgur.com/qPHKDh5
https://imgur.com/wMIp2RZ
https://imgur.com/dukDy2M
https://imgur.com/QDm0URg

</p></details> 

 
This has my GM9 trim and untrim also my flash dump
https://ufile.io/q7k4w

Any help to get this to work would be great thanks

**Comment by zoogie** [comment]
From my initial examination, that looks to be identical to rev6:

https://github.com/kitling/flashcart_core/issues/9#issuecomment-325106067
ntr_blowfish_retail = 0x0 (plain)
b9s_payload = 0x80000 (enc)
b9s_header = 0x1FFE00 (enc)
enc = same memecrypto as latest r4ids.cn versions

Your backup.bin also has ntrboot injected for the latest r4igoldcn versions (i.e. b9s header at EE00), which of course won't work in your case. Hope you have a clean backup somewhere before you attempted to flash it (might be able to use a clean rev6 if you're lucky).

**Comment by Leadger69** [comment]
No unfortunately I can't seem to find my clean backup...so I guess that mean I'm screwed

**Comment by zoogie** [comment]
You should still be able to flash ntrboot provided we can figure out your card. Just going back to running your card as a DS flashcard MIGHT be off the table. Read that last sentence of the OP.

I'll try to make a backup.bin later for you to test as an ntrboot card if you're willing.

**Comment by Leadger69** [comment]
I'll test anything..i don't mind

**Comment by zoogie** [comment]
[backup.zip](https://github.com/kitling/flashcart_core/files/1370098/backup.zip)

Alright, try that. Just put the file in sdmc://ntrboot/ and select "restore flash" with the following tool:
http://s000.tinyupload.com/index.php?file_id=00664373029281777476
Don't choose "inject ntrboot" under any circumstance and make sure to name your payload ntrboot.firm

**Comment by Leadger69** [comment]
It detected it fine as r4i gold 3ds and backed up fine as well..but it doesn't load as a ds flashcart

**Comment by zoogie** [comment]
@Leadger69 - that backup.bin was intended to be an ntrboot flash, not a DS flashcard restoration. The icon will show up as blank on home menu.

**Comment by Leadger69** [comment]
Sorry my bad...but it still didn't  work no safeboot

**Comment by zoogie** [comment]
Ok, try this one:
[backup.zip](https://github.com/kitling/flashcart_core/files/1370192/backup.zip)

Same directions as the last one.

**Comment by Leadger69** [comment]
nope still no boot

**Comment by zoogie** [comment]
What exactly is happening? Is the system turning off, booting to home menu etc. Did you rename safeb9sinstaller.firm (or some other payload firm) to ntrboot.firm and to do the correctkey combination + magnet?

Anyway, if you did all that and still no luck, you could try restoring DS flashcard compat by reflashing this.
http://s000.tinyupload.com/index.php?file_id=04706518342023310360
Let me know if you get some icon and if it boots (you will need a hacked 3ds or maybe dsi to boot it).

**Comment by Leadger69** [comment]
it stays off untill i move the magnet then boots home screen.also does the sd card need to be in the flashcart and I thought that you rename the safe.firm to boot.firm

**Comment by zoogie** [comment]
No, you need to rename safeb9sinstaller.firm to ntrboot.firm as I said above. The microsd does not need to be inside the flashcard for ntrboot. The 3ds system itself needs an sd card of course.

**Comment by Leadger69** [comment]
Sorry for the no response I was at work I tried all the things in the op but still no safe boot..so I tried the other file and now the card is back to normal with screen and boot like a regular d's flash cart on my backed 3ds

**Comment by zoogie** [comment]
Well that's certainly good news. At least we know that rev6 and rev7 should be essentially identical. Sorry we couldn't get ntrboot to work on your rev7, but we'll keep trying.

**Comment by Leadger69** [comment]
well at least I'm helping to figure stuff out

**Comment by zoogie** [comment]
@Leadger69 please flash 
https://github.com/kitling/flashcart_core/issues/75#issuecomment-335318958
again.

boot with the ntrboot combo, then immediately run godmode9 and dump MEMORY VIRTUAL -> arm9.bin. should be a 1MB file. Upload that please.

It's really difficult for me to debug this without having the cart on hand.

**Comment by Leadger69** [comment]
Sorry but what do you mean by ntrboot combo

**Comment by zoogie** [comment]
Sleep + Start + Select + X then press and hold power

**Comment by zoogie** [comment]
Try and use this version of godmode9
https://github.com/d0k3/GodMode9/releases/download/v1.2.8/GodMode9-20170729-114727.zip

rename it to boot.firm and put it on the sd card

**Comment by Leadger69** [comment]
Oh so do that on my stock 3ds then put the flash card back into my cfw 3ds and dump the arm9.bin

**Comment by zoogie** [comment]
No everything needs to be done on the cfw one.

**Comment by Leadger69** [comment]
there is a arm9.mem but no arm9.bin

**Comment by zoogie** [comment]
yeah it's arm9.mem forgot

**Comment by Leadger69** [comment]
https://ufile.io/nm2r0 there you go

**Comment by zoogie** [comment]
Alright, I've got to go to bed but I'll look at it later today and get back with you.

**Comment by Leadger69** [comment]
Sounds I'm gone to bed too night shift

**Comment by zoogie** [comment]
@Leadger69 it looks like you don't have any ds card inserted :/
No trace of a header anywhere.

Did you
1. restore the backup.bin I mentioned above.
https://github.com/kitling/flashcart_core/files/1370192/backup.zip
2. Rename that particular godmode9 I linked to boot.firm and put it on the sd card root.
https://github.com/d0k3/GodMode9/releases/download/v1.2.8/GodMode9-20170729-114727.zip
3. insert your r4iGold into the system.
4. ntrboot into godmode9 and dump arm9.mem.

The only other issue I can think of is contact issues. Make sure the contacts are clean and a dump doesn't read zero's or whatever.

**Comment by Leadger69** [comment]
https://ufile.io/1ap9r how about this one I think the magnet was in the wrong place

**Comment by zoogie** [comment]
Still no trace of a DS header.

Lets try a different track.
[backup.zip](https://github.com/kitling/flashcart_core/files/1385917/backup.zip)
Restore this backup.bin and tell me if you can boot the card in ds mode on your cfw system.

**Comment by Leadger69** [comment]
i backed it up but when i put it in my 3ds it freezes and wont turn on

**Comment by zoogie** [comment]
Damn, it was worth a shot.
Thanks for testing, will let you know if I can think of anything else to try.

**Comment by Leadger69** [comment]
alright sounds good..thanks for the help

**Comment by zoogie** [comment]
@Leadger69 
Another little test if you're still around.
Make sure https://github.com/kitling/flashcart_core/files/1370098/backup.zip is restored to your card then run https://transfer.sh/OyjwV/fctester.firm
That will dump ntrboot.nds to your sd card root. Please upload that if you can.

**Comment by Leadger69** [comment]
I restored the backup then ran fctester.firm then my cfw 3ds says "ntrcard_begin_key1 failed: -1 0 7EAA0AB2 press B to exit

**Comment by zoogie** [comment]
More testing if you're up to it.

Ok, make sure this backup
https://github.com/ntrteam/flashcart_core/issues/75#issuecomment-335318958
is flashed (if no icon appears then you don't have to flash anything)

then run
https://github.com/ntrteam/dsfirmverify/releases/tag/v0.0.3
unfortunately you need a ds lite or phat for this to work and two ds flashcards lol.
One card could work but you'd need to flash your ds mode to the r4iGold with the 3ds, run this dsfirmverify, then flash the ntrboot backup.bin with 3ds, then complete the hotswap and press A. That's a lot to ask I know :/

And tell me the result displayed.

**Comment by Leadger69** [comment]
I did it and it says ncgc_nbegin_key1 failed: -2 0xDC306FB7 0xF132C84E
don't know what that means..lol

**Comment by Leadger69** [comment]
I did it and it says ncgc_nbegin_key1 failed: -2 0xDC306FB7 0xF132C84E
don't know what that means..lol

**Comment by zoogie** [comment]
I don't know either. Maybe @angelsl would know but he has his hands full with other stuff most likely.

btw - https://gbatemp.net/threads/r4igold-r4ids-cn-rev-6-7-8-offering-ntrboot-ak2i-or-20-eshop.488295/
tl;dr I'm looking to buy a rev 6-8 r4ids.cn off of somebody. I honestly think this is the only way I'll be able to figure this card out. :/

**Comment by angelsl** [comment]
@Leadger69

Did you remove and reinsert the card fully when it told you to do so?

The two hex numbers are the chipid it got from raw and from Key1 mode. Both look like gibberish, so..

**Comment by Leadger69** [comment]
I'll give it a try again later and post the findings

**Comment by Leadger69** [comment]
I'm getting my friends acekard 2 version 2.1 tomorrow so I'll try the 2 flash cart method hopefully this will work...lol

**Comment by twocows** [comment]
Similar situation here. I was able to revert it back to a working flashcart since I already have another hacked 3DS, though. Looks like the cart I have is an rev 7 since that was the backup that was able to get it in working shape again (and coincidentally the first one I tried).

zoogie, did you still need a card? If so, are you located in the US?

**Comment by zoogie** [comment]
@twocows 
I might. Right now I should be able to have a rev6 on Monday.
Give me a couple of days after that to see if I can get anything working and produce something for you guys to test.

**Comment by twocows** [comment]
Actually, this is my bad, I didn't realize this was for the revision 7 non-RTS version. I'm using the revision 7 RTS version. It's possible this is just user error on my part (in my defense, I did try for like an hour), so I'll have to play around some more.

**Comment by zoogie** [comment]
@twocows a rev7 is a rev7 as long as r4ids.cn is the url on the front sticker and a "7" on the back connector pin.

**Comment by zoogie** [comment]
@twocows 
Yeah that's definitely not a rev7. That's an A7, something totally different. (i think those are rev9 or C).

**Comment by Leadger69** [comment]
Did it again and it says ncga_nbegin_key1 failed:-2 0x00000000 0x0E112A02 don't know if this helps

**Comment by zoogie** [comment]
@Leadger69 
Got it fixed on rev6 (#9) so it should work with this rev. as well.
https://gist.github.com/zoogie/bc4edb08c8e9fce6276b22562424f9b0

If you can't figure out how to use that script I can probably provide you with a binary to test tommorrow (tues).

**Comment by nikhiljha** [comment]
> 0x0E112A02

Old issue, but I managed to get this too by flashing my card (R4i-sdhc 3DS wifi non rts) with the gold card option.

I did this by hotswapping the card on my DSi, which probably messed something up.

Just another data point for any future people

**Comment by Bolt23** [comment]
@nikhiljha How did you HOTSWAP? I mean what's the process? I get the same error when I try it on my R4i 1.4 (www.ndsiLL.com) BLUE (probably a SDHC) #119 Help a mate from same country will ya?

**Comment by nikhiljha** [comment]
Don't do it, it messes things up further by reading from the wrong memory. :)

**Comment by Bolt23** [comment]
Okay but can you explain to me what is hotswapping, and how did you fix this, I have the exact same issue please be specific and detailed. Thanks in Advance

**Comment by rblacha** [comment]
Hi!
I need some help for my R4i 3ds (non RTS) Rev 8.
I have forgotten to restore my Flashhcard before i started sd cleanup... 
My Backup is now deleted and i can not find a other Backup thath works on my Flashcard.
Many Links here and in other Forums are not availebel (404 error or File is delet)
Maybe is here someone, that have/can dump his flashcard and upload the Backup.bin

Excuse me, I know my english is bad :-)

Thanks...

![img-20180714-wa0001](https://user-images.githubusercontent.com/41227748/42722372-37e422a0-874b-11e8-9cc0-869cc7d2b63c.jpg)
![img-20180714-wa0000](https://user-images.githubusercontent.com/41227748/42722374-3813f3b8-874b-11e8-9dfb-9442d14c44e4.jpg)

**Comment by rblacha** [comment]
Backup.bin for R4i 3ds (non RTS) Rev 8

Yeah, I found it...
I have found my Backup on a other sd card :-) sorry...
Here, maybe it help someone else...

[R4iGold3DS-backup.zip](https://github.com/ntrteam/flashcart_core/files/2194798/R4iGold3DS-backup.zip)
